npm/pkg/controlplane/controllers/v1/networkPolicyController.go

```diff
@@ -259,6 +266,11 @@ func (c *NetworkPolicyController) syncNetPol(key string) error {
 	// If DeletionTimestamp of the netPolObj is set, start cleaning up lastly applied states.
 	// This is early cleaning up process from updateNetPol event
 	if netPolObj.ObjectMeta.DeletionTimestamp != nil || netPolObj.ObjectMeta.DeletionGracePeriodSeconds != nil {
+		if _, ok := c.rawNpMap[key]; ok {
+			// record time to delete policy if it exists (can't call within cleanUpNetworkPolicy because this can be called by a pod update)
+			timer := metrics.StartNewTimer()
+			defer metrics.RecordPolicyApplyTime(timer, metrics.DeleteMode)
```

Comment:
Out of curiosity what happens if we error out while performing deletion? The time elapsed will skew the metrics, will it not, when compared to successful deletes, unless we can correlate the two events? 
My concern is when we aggregate metrics over a period of time to get a sense of deletion times (for performance statistic), if we were erroring out we would have short deletion times, which might inadvertently convey that we are performing well in deletions since we do not have the error context associated with the captured time metric.
Something to think about. WDYT? /cc @vakalapa @matmerr

Reply:
Maybe an extra field that passes the success state when `RecordXYZApplyTime(...)` is called?

Reply:
In the past we've recorded regardless of success. NPM rarely fails unless there's an IcM-worthy problem if I'm not wrong @matmerr, @vakalapa ?

An error or two shouldn't really affect an overall average of say 30+ adds I think. Those quick runs also wouldn't impact the quantiles (0.5, 0.9, 0.99).

Reply:
Agreed, the error percent will be low, so the results wont be skewed way too much. Good thing in V2 we are logging faulre counts too. Lets keep V1 simple and enhance V2 with such error check logic as separate effort.

Reply:
Incorporated this suggestion. Now metrics have an "error" label